raft/raft.go

```diff
@@ -961,6 +1054,20 @@ func (r *raft) Step(m pb.Message) error {
 			r.hup(campaignElection)
 		}
 
+	case pb.MsgStorageAppendResp:
+		if m.Index != 0 {
+			r.raftLog.stableTo(m.Index, m.LogTerm)
+		}
+		if m.Snapshot != nil {
+			r.appliedSnap(m.Snapshot)
+		}
+
+	case pb.MsgStorageApplyResp:
+		r.appliedTo(m.Index)
+		// NOTE: we abuse the LogTerm field to store the aggregate entry size so
+		// that we don't need to introduce a new field on Message.
+		r.reduceUncommittedSize(m.LogTerm)
```

Comment:
Anchoring a comment here because there's an active discussion about this elsewhere.

Reply:
Done.